teams/adcs/LSM303AGR_magnetometer.c

```diff
@@ -0,0 +1,268 @@
+/*
+ *  LSM303AGR_magnetometer.c
+ *  Created on: Jan 15, 2020
+ *  Author: Shubhkarman Singh
+ */
+
+#include <stdint.h>
+#include <math.h>
+#include "LSM303AGR_magnetometer.h"
+
+#define MAX_BUFF_SIZE   0x25
+#define MAX_CALIBRATION_SAMPLES 15
+#define CALIBRATION_IGNORE_SAMPLES 5
+#define MAX_NUM_MAGNETOMETERS 2 // one for each i2c bus
+#define MIN_VALID_MAG_READING -2048 //0xF800
+#define MAX_VALID_MAG_READING 2047 //0x07FF
+
+#define NORMAL_OPERATION 1
+#define SELF_TEST_OPERATION 0
+
+FILE_STATIC uint16_t SELF_TEST_NOMINAL_X = MAG_HMC5883L_SELF_TEST_GAUSS_FACTOR_X * MAG_HMC5883L_GAIN_1370_LSB_GAUSS;
+FILE_STATIC uint16_t SELF_TEST_NOMINAL_Y = MAG_HMC5883L_SELF_TEST_GAUSS_FACTOR_Y * MAG_HMC5883L_GAIN_1370_LSB_GAUSS;
+FILE_STATIC uint16_t SELF_TEST_NOMINAL_Z = MAG_HMC5883L_SELF_TEST_GAUSS_FACTOR_Z * MAG_HMC5883L_GAIN_1370_LSB_GAUSS;
+
+#if defined(__HIL_AA_GLITCHFILTER__)

```

Comment:
Are we planning to do any HIL testing?  Is there any other reason why we might need this glitch filter?